pkg/querier/tail.go

```diff
@@ -264,25 +279,48 @@ func (t *Tailer) getCloseErrorChan() <-chan error {
 	return t.closeErrChan
 }
 
+func (t *Tailer) recordStream(id uint64) {
+	t.seenStreamsMtx.Lock()
+	defer t.seenStreamsMtx.Unlock()
+
+	if _, ok := t.seenStreams[id]; ok {
```

Comment:
If you're only writing once there's a lot of chances that `sync.Map` will perform better. Not sure this is important in this case but at least I shared the options for later.